Integrations/integration-Cymon.yml

```diff
@@ -0,0 +1,481 @@
+
+    USE_SSL = True if demisto.params().get('unsecure') else False
+
+    DEFAULT_HEADERS = {
+      "Content-Type": "application/json"
+    }
+
+    ''' HELPER FUNCTIONS '''
+
+    def http_request(method, url, headers):
+        LOG('running %s request with url=%s\theaders=%s\n' % (method, url, headers))
+
+        try:
+            res = requests.request(method,
+                                    url,
+                                    verify=USE_SSL,
+                                    headers=headers)
+
+            if res.status_code == 200:
+                return res.json()
+            # 204 HTTP status code is returned when api rate limit has been exceeded
+            elif res.status_code == 204:
+                return_error("You've reached your API call quota.")
+            elif res.status_code == 404:
+                return {}
+
+            res.raise_for_status()
+
+        except Exception, e:
+            LOG(e)
+            raise(e)
+
+    ''' DOMAIN COMMAND '''
+    def get_domain_report(domain):
+        reports = {}
+
+        from_param = 0
+        size_param = 10
+        total = None
+
+        url = '{}/{}/{}?from={}&size={}'.format(SERVER_URL_V2, 'domain', domain, from_param, size_param)
+
+        while total is None or total > from_param:
+            response = http_request('GET', url, DEFAULT_HEADERS)
+
+            hits = response.get('hits', [])
+            for hit in hits:
+                title = hit.get('title')
+                timestamp = datetime.datetime.strptime(hit.get('timestamp'), '%Y-%m-%dT%H:%M:%S.%fZ')
+
+                if (title in reports and reports.get(title).get('timestamp') < timestamp) or title not in reports:
+                    reports.update({title: {
+                        'timestamp': timestamp,
+                        'reported_by': hit.get('reported_by'),
+                        'feed': hit.get('feed'),
+                        'tags': hit.get('tags'),
+                        'ioc': hit.get('ioc'),
+                        'location': hit.get('location', {}).get('country')
+                    }})
+
+            from_param = from_param + size_param
+            total = int(response.get('total', 0))
+
+            url = '{}/{}/{}?from={}&size={}'.format(SERVER_URL_V2, 'domain', domain, from_param, size_param)
+
+        report_results = []
+
+        for report in reports:
+            report_results.append({
+                'Title': report.title(),
+                'Feed': reports.get(report).get('feed'),
+                'Timestamp': reports.get(report).get('timestamp').strftime("%Y-%m-%d %H:%M:%S"),
+                'Tags': reports.get(report).get('tags'),
+                'Hostname': reports.get(report).get('ioc').get('hostname'),
+                'IP': reports.get(report).get('ioc').get('ip'),
+                'Domain': reports.get(report).get('ioc').get('domain'),
+                'Reported By': reports.get(report).get('reported_by'),
```

Comment:
all reports have these attributes? no possibility for nil error here?

Reply:
Didn't came across such case, but will address it anyway